### DISCUSSION TITLE
[BUG] Perl include path @INC on Windows-latest contains only linux paths

### DISCUSSION THREAD

**Discussion by ldorau** [community/community#26175] [Actions]
<p>BUG: Perl include path <span class="mention">@INC</span> <strong>on Windows-latest</strong> contains <strong>only Linux paths</strong> :</p>
<p>A message from <strong>Windows-latest</strong> :</p>
<p>“Can’t locate Text/Diff.pm in <span class="mention">@INC</span> (you may need to install the Text::Diff module) (<span class="mention">@INC</span> contains: <strong>/usr/lib/perl5/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/core_perl /usr/share/perl5/core_perl</strong> )”</p>
<p>See: <a href="https://github.com/ldorau/pmdk/commit/baf51f10f7de53fa09549e0a27cc0e1e57658520/checks?check_suite_id=337927144" rel="nofollow noopener">https://github.com/ldorau/pmdk/commit/baf51f10f7de53fa09549e0a27cc0e1e57658520/checks?check_suite_id=337927144</a></p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>In github hosted windows runners, it don’t add Perl to Path environment variable. I checked your workflow, you have added a step with echo “::add-path::$Env:perl” to add Perl to Path. This can resolve this issue.   </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/ee6c64dd043a6f0311fc03b82259a9dd8c959811.png" data-download-href="/uploads/short-url/y1bS0g71CsUcIn1nXCqDA5z9zIl.png?dl=1" title="Perl path.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095932-deaecc44-6b38-4648-b011-8dde102891c8.png" alt="Perl path.png" data-base62-sha1="y1bS0g71CsUcIn1nXCqDA5z9zIl" width="690" height="360" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/ee6c64dd043a6f0311fc03b82259a9dd8c959811_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Perl path.png</span><span class="informations">888×464 11.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ldorau** [comment]
<p>Yes, adding Perl to Path resolves the issue, but anyway <span class="mention">@INC</span> should not contain linux paths on a windows runner…</p>

**Comment by hu13** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> wondering if you still have a working workflow for this.</p>
<p>I tried the suggested solution, but no luck.</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/hu13">@hu13</a> It would be better if you could share your detail workflow yaml and logs in a new ticket . </p>
